### PR TITLE
Support alternative graphics drivers

### PR DESCRIPTION
**Pull request by joe-singleton-itdev**
* Add `--graphics-driver` option to create and update parsers
* Set/update graphics driver in VM config and libvirt config
* Rename driver` option to `--hdd-driver` in create and update parsers

### CODE REVIEW

**Comment by itdevreviewboard** [comment]
Can one of the MCVirt admins verify this patch?

File: source/usr/lib/mcvirt/virtual_machine/factory.py [hunk 1]
```diff
@@ -118,6 +130,11 @@ def checkName(self, name, ignore_exists=False):
 
         return True
 
+    def checkGraphicsDriver(self, driver):
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/virtual_machine/factory.py]: Sorry to be a pain, the coding standard specifies that we should be using snake_case for methods (although most of the code in this class pre-exists this and hasn't been converted, so not your fault at all. Just be good to use this for all new code

File: source/usr/lib/mcvirt/virtual_machine/factory.py [hunk 2]
```diff
@@ -149,6 +166,12 @@ def _create(self, name, cpu_cores, memory_allocation, hard_drives=[],
         if hard_drive_driver is not None:
             HardDriveDriver[hard_drive_driver]
 
+        # If no graphics driver has been specified, set it to the default
+        if graphics_driver is None:
+            graphics_driver = self.DEFAULT_GRAPHICS_DRIVER
+        # Check the driver name is valid
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/virtual_machine/factory.py]: Could you add a blank line before comments?

File: source/usr/lib/mcvirt/virtual_machine/virtual_machine.py [hunk 3]
```diff
@@ -1284,3 +1285,34 @@ def updateXML(domain_xml):
                 os_xml.append(new_boot_xml_object)
 
         self._editConfig(updateXML)
+
+    @Pyro4.expose()
+    def updateGraphicsDriver(self, driver):
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/virtual_machine/virtual_machine.py]: Could you update this one as well?

**Comment by MatthewJohn** [comment]
:shipit:

**Pull request merged**